core/src/test/scala/org/apache/spark/network/netty/NettyBlockTransferTimeoutSuite.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.netty
+
+import scala.concurrent.duration._
+
+import io.netty.handler.timeout.ReadTimeoutException
+import org.mockito.Mockito._
+import org.mockito.Matchers.any
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.scalatest.FunSuite
+
+import org.apache.spark.{SparkConf, SecurityManager}
+import org.apache.spark.network.BlockDataManager
+import org.apache.spark.network.buffer.ManagedBuffer
+import org.apache.spark.storage.ShuffleBlockId
+import org.scalatest.concurrent.Timeouts
+
+class NettyBlockTransferTimeoutSuite extends FunSuite with Timeouts {
+  test("read timeout") {
```

Comment:
This regression test demonstrates the problem that this patch aims to fix.  This test has a client attempt to fetch from a block transfer service which has been configured to stall for a long period of time. The block fetch should fail with a read timeout exception, which has been configured to be relatively short.